client/snap_op.go

```diff
@@ -314,20 +314,31 @@ type snapRevisionOptions struct {
 }
 
 type downloadAction struct {
-	SnapName string `json:"snap-name,omitempty"`
+	SnapName    string `json:"snap-name"`
+	HeaderPeek  bool   `json:"header-peek,omitempty"`
+	ResumeToken string `json:"resume-token,omitempty"`
+
 	snapRevisionOptions
 }
 
 type DownloadInfo struct {
 	SuggestedFileName string
 	Size              int64
 	Sha3_384          string
+	ResumeToken       string
+}
+
+type DownloadOptions struct {
+	SnapOptions
+
+	HeaderPeek  bool   `json:"header-peek,omitempty"`
```

Comment:
this doesn't look like it is sent to the daemon directly, so not sure we need the json tags